src/components/CommentsList.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { CommentsListItem } from './CommentsListItem';
+import { Comment } from '../types/Comment';
+import { HandleCommentDelete } from '../types/handlers';
+
+type Props = {
+  comments: Comment[];
+  onCommentDelete: HandleCommentDelete;
+};
+
+export const CommentsList = React.memo(
+  ({ comments, onCommentDelete }: Props) => {
+    const [error, setError] = useState(false);
+
+    const handleCommentDeleteClick = async (commentId: number) => {
+      setError(false);
+      setError(!(await onCommentDelete(commentId)));
```

Comment:
Using exceptions and rethrowing errors can be a better approach for handling errors as it allows you to provide more detailed information about what went wrong. Instead of returning a boolean, you can throw an error in the `onCommentDelete` function and catch it in the `handleCommentDeleteClick` function.